src/components/button/button.svelte

```diff
@@ -76,10 +76,26 @@
   class:isTiny={size === 'tiny'}
   class:fab
   class:isLoading
+  disabled={isLoading || isDisabled || undefined}
   on:click={onClick}
   {...$$restProps}
 >
-  <slot>Leo Button</slot>
+  {#if isLoading}
+    {#if $$slots.loading}
+      <slot name="loading" />
+    {:else if !fab}
+      <div>
+        <slot>Leo Button</slot>
+      </div>
+    {/if}
+    <ProgressRing />
+  {:else}
+    <slot name="left-icon" />
+    <div>
+      <slot>Leo Button</slot>
+    </div>
+    <slot name="right-icon" />
+  {/if}
```

Comment:
I think having both icon slots conditional on `loading` is going to be a bit strange when the button is changing width, but if that's the design then I guess we'll see how it feels in real use.